src/app.js

```diff
@@ -0,0 +1,51 @@
+import axios from 'axios';
+
+const searchForm = document.getElementById('search-form');
+searchForm.addEventListener('submit', searchCountry);
+
+const countryInfoBox = document.getElementById('search-result');
+const errorMessageBox = document.getElementById('error-message');
+
+function searchCountry(e) {
+    e.preventDefault();
+    const queryfield = document.getElementById('query-field');
+    fetchCountryDetails(queryfield.value);
+    queryfield.value = '';
+}
+
+async function fetchCountryDetails(name) {
+    countryInfoBox.innerHTML = ``;
+    errorMessageBox.innerHTML = ``;
+
+    try {
+        const result = await axios.get(`https://restcountries.com/v2/name/${name}`);
+        const country = result.data[0];
```

Comment:
Je zou hier nog spreading kunnen toepassen, zodat je niet overal `country.` voor hoeft te zetten
```suggestion
        const { name, capital, subregion, population, currencies } = result.data[0];
```